rfcs/text/0018_timeslider.md

```diff
@@ -0,0 +1,217 @@
+
+The main effort to support efficient data-fetch in a maps-context is to use `.mvt` as the default data format ([https://github.com/elastic/kibana/issues/79868](https://github.com/elastic/kibana/issues/79868)). This is a stack-wide effort in collaboration with the Elasticsearch team ([https://github.com/elastic/elasticsearch/issues/58696](https://github.com/elastic/elasticsearch/issues/58696), which will add `.mvt` as a core response format to Elasticsearch.
+
+Growing the use of `mvt`([https://docs.mapbox.com/vector-tiles/reference/](https://docs.mapbox.com/vector-tiles/reference/)) in Maps will help with both pre-fetching and client-side caching:
+- `mvt` is a binary format which allows more data to be packed inside, as compared to Json. Multiple tiles are patched together, so this introduces a form of parallelization as well. Due to growing the amount of data inside a single tile, and due to the parallelization, Maps has a pathway to increase the number of features that can be time-filtered.
+- Because vector tiles have fixed extents and scales (defined by a `{x}/{y}/{scale}`-tuple), this type of data-fetching allows tiles to be cached on the client. This cache can be the implicit browser disc-cache, or the transient in-mem cache of mapbox-gl. Using mvt thus provides a pathway for fast toggling back&forth between timeslices, without round-trips to fetch data.
+
+
+##### 2.3.3.4 timeslider and async search
+
+It is unclear on what the practical uses for async-search would be in the context of a timeslider-control in Maps. 
+
+Timeslider is a highly interactive control that require immediate visual feedback. We also do not intend to activate timeslider in Maps on a Dashboard (see above).
+
+End-users who need to view a dashboard with a long-running background search will need to manipulate the _global Kibana time picker_ to select the time-range, and will not be able to use the timeslider to do so.
+
+# 3. Unresolved questions
+
+## Making Timeslider a Kibana Embeddable
+
+This below is a forward looking section. It is a proposal of how the Timeslider-UI can be exposed as an Embeddable, when that time should come.
+
+We expect a few steps:
+- This would require the extraction of the timeslider React-component out of Maps into a separate plugin. As outlined above, this migration should be fairly straightforward, a "cut and paste".
+- It would require the creation of a `TimesliderEmbeddable` which wraps this UI-component.
+- It would also require the introduction of a new piece of embeddable-state, `Timeslice`, which can be controlled by the `TimesliderEmbeddable`. 
+We believe it is important to keep `timeslice` and `timerange` separate, as individual apps and other embeddables will have different mechanism to efficiently fetch data and respond to changes in `timeslice` and/or `timerange`.
```

Comment:
I'm not entirely convinced of this, but I don't think it has any effect on this RFC, which looks great to me. Since I don't think the current implementation plan will back us into any corners one way or the other, LGTM!

Reply:
thx @stacey-gammon 

Yes, we can spin-off the discussion of adding new embeddable state. It agree it might  not be necessary from a high level (e.g. timeslider as an Embeddable).

The reason we raised it here, is because keeping `timeslice` and `timerange` separate does solve a particular problem for Maps. It allows some layer-types to _prefetch_ data: ie. get all the data within the entire `timerange` up-front, and then do client-side filtering to respond to timeslice-changes. This avoids round-trips to Elasticsearch. But this is definitely an optimization.